packages/bot/src/stores/chart-store.js

```diff
@@ -0,0 +1,123 @@
+
+import {
+    action,
+    computed,
+    observable,
+    reaction,
+}                     from 'mobx';
+import { tabs_title } from '../constants/bot-contents';
+import ServerTime     from '../services/api/server_time';
+
+const g_subscribers_map = {};
+let WS;
+
+export default class ChartStore {
+    constructor(root_store) {
+        this.root_store = root_store;
+        WS = root_store.ws;
+        const { run_panel } = root_store;
+
+        reaction(() =>
+            run_panel.is_running, () =>
+            run_panel.is_running ?
+                this.startBot() :
+                this.stopBot()
+        );
+    }
+
+    @observable symbol;
+    @observable is_chart_loading;
+    @observable main_barrier = null;
+    @observable chart_type;
+    @observable granularity;
+
+    @computed
+    get is_contract_ended() {
+        const { transactions } = this.root_store;
+
+        return transactions.contracts.lenght > 0 && transactions.contracts[0].is_ended;
+    }
+
+    @action.bound
+    startBot() {
+        this.updateSymbol();
+
+        const { main_content } = this.root_store;
+        main_content.setActiveTab(tabs_title.CHART);
+    }
+
+    @action.bound
+    stopBot() {
+        const { main_content } = this.root_store;
+        main_content.setActiveTab(tabs_title.WORKSPACE);
+    }
+
+    @action.bound
+    updateSymbol() {
+        const workspace = Blockly.derivWorkspace;
+        const market_block = workspace.getAllBlocks().find(block => {
+            return (block.type === 'trade_definition_market');
+        });
+
+        const symbol = market_block.getFieldValue('SYMBOL_LIST');
```

Comment:
- It would be good to add an `if` condition here, because users may not always have the `market_block` in their workspace.
- In some cases the returned value may be an empty string (`""`) or `na`, this should be handled also.